docs/proposals/registry/registry-structure.md

```diff
@@ -0,0 +1,101 @@
+
+This design proposal is a follow up to [Devfile Registry Packaging](https://github.com/devfile/api/blob/master/docs/proposals/registry/devfile-packaging.md) and I recommend reading that first.
+
+## Terminology
+
+Some of the following terms will be used throughout this design proposal:
+
+**Devfile Registry Repository:** The GitHub repository that hosts the devfile stacks for consumption within an OCI registry. For example, [devfile/registry](https://github.com/devfile/registry).
+
+**Devfile Index Image:** The container image containing the devfile stacks and index.json used to bootstrap the OCI registry with devfile stacks
+
+**Registry Build:** The process that takes the devfile registry repository, generates the index.json and builds it up into the devfile index container image.
+
+**Registry Bootstrap:** The process that pushes the devfile stacks on the devfile index container to the OCI registry.
+
+## As-is Today
+Currently, the top-level structure of a devfile registry’s repository is unwritten, but it usually consists of a **devfiles** or **stacks** folder, and any associated files specific to that registry
+
+[devfile/registry](https://github.com/devfile/registry):
+<img width="914" alt="Screen Shot 2020-11-30 at 2 12 13 PM" src="https://user-images.githubusercontent.com/6880023/100653219-12a48100-3316-11eb-949c-38073a19acbc.png">
+
+Inside each **stacks** or **devfiles** folder, each folder corresponds to a devfile stack, containing usually just the devfile.yaml and a meta.yaml:
+<img width="1232" alt="Screen Shot 2020-11-30 at 2 10 50 PM" src="https://user-images.githubusercontent.com/6880023/100653095-e2f57900-3315-11eb-8c3f-86e56896ef15.png">
+
+<img width="1229" alt="Screen Shot 2020-11-30 at 2 11 20 PM" src="https://user-images.githubusercontent.com/6880023/100653133-f0aafe80-3315-11eb-8393-dbabfd8ce0d6.png">
+
+As part of the registry build, the index.json is generated based off of the stacks in the repository, and a devfile index image is generated containing the index.json and stacks.
+
+When deploying the OCI registry, the registry bootstrap process parses the index.json to find the devfile.yaml for each stack, and pushes the devfile up to the registry as a single layer. No other stack artifacts are pushed up as part of the layer, or as separate layer.
+
+## Problem
+Our approach of pushing only the devfile.yaml works fine currently because the stacks we’re pushing only have a devfile.yaml in them. However, devfile stacks may contain more than _just_ the devfile, and may also contains resources that are re-used across multiple stacks (such as certain vsx plugins).
+
+We need a defined way of knowing what files in a stack to push up in a, what layers each file in the stack should belong to, and where the stack's files should located. 
+
+## Proposal
+
+To solve the issue listed above, we should:
+1) Formally the expected structure of the Devfile Regisry Repository.
```

Comment:
minor typos:
```suggestion
1) Formalize the expected structure of the Devfile Registry Repository.
```

Reply:
Gah, thanks. The `t` key on my laptop has recently started only working like half the time, so I've had lots of `regisry` -> `registry` corrections to do in the past week haha.

Reply:
We're watching technology degrade in real-time during this pandemic ;) I need to mash my spacebar to get it to register.